lib/chef/util/windows/net_user.rb

```diff
@@ -96,8 +96,13 @@ def validate_credentials(passwd)
     token = Security.logon_user(@username, nil, passwd,
                LOGON32_LOGON_NETWORK, LOGON32_PROVIDER_DEFAULT)
     true
-  rescue Chef::Exceptions::Win32APIError
-    false
+  rescue Chef::Exceptions::Win32APIError => e
+    if e.to_s.include? "The user name or password is incorrect"
+      return false
+    end
+    # all other exceptions will assume we cannot logon for a different reason (e.g. an Account Restriction)
+    Chef::Log.trace("Unable to login with the specified credentials. Assuming the credentials are valid.")
```

Comment:
I have a feeling some day we're going to find we're swallowing an exception here that we're interested in. Since this is already waaay down in trace, could we add a `Chef::Log.trace(e.msg)` after this?

Reply:
Addressed.